libshvcore/src/utils/shvfilejournal.cpp

```diff
@@ -757,27 +762,50 @@ chainpack::RpcValue ShvFileJournal::getLog(const ShvFileJournal::JournalContext
 						continue;
 					logDShvJournal() << "\t\t MATCH";
 				}
-				if(params_since_msec > 0 && e.epochMsec < params_since_msec) {
-					if(params.withSnapshot) {
-						if(e.sampleType == ShvJournalEntry::SampleType::Continuous) {
-							last_snapshot_msec = e.epochMsec;
-							addToSnapshot(snapshot, e);
+				this_file_values[e.path] = e;
+				entries_to_write[0] = e;
+				entries_to_write.resize(1);
+				if(rd.isInSnapShot()) {
+					prev_file_values.erase(e.path);
+				}
+				else if(!prev_file_values.empty()) {
+					for(const auto &kv : prev_file_values) {
+						ShvJournalEntry e3 = kv.second;
+						if(!e3.value.hasDefaultValue() && this_file_values.find(e3.path) == this_file_values.cend()) {
+							// value is set in previous file, but it is not present in current file snapshot
+							// this can happen if device was switched off and for example error set to true
+							// was cleared meanwhile
+							// we have to inject this lost information into the current file snapshot
+							e3.value.setDefaultValue();
+							entries_to_write.push_back(kv.second);
```

Comment:
Myslim ze tady se do e3 musi nastavit cas m_snapShotEpochMsec. Pro funkci `write_snapshot` by to nevadilo, ale mohlo by to vadit pri porovnavani na radce 786 a ve funkci `append_log_entry`.

Reply:
spravna pripominka :+1:

Reply:
opraveno